src/zenml/zen_stores/sql_zen_store.py

```diff
@@ -1377,11 +1378,10 @@ def migrate_database(self) -> None:
                     SQLModel.metadata.create_all  # type: ignore[arg-type]
                 )
             with Session(self.engine) as session:
-                session.add(
-                    IdentitySchema(
-                        id=str(GlobalConfiguration().user_id).replace("-", "")
-                    )
+                deployment_id = os.getenv(
```

Comment:
@stefannica I'm unsure whether this is the right place to use this environment variable. When implemented like this, the environment variable will only be respected when deploying the server for the first time. We could use it to have precedence over the DB entry by checking for the env variable existence in `SQLZenStore.get_deployment_id()`. What do you think?

Reply:
I suggest not adding another yet environment variable for this, it just complicates this even more. We already have `ZENML_USER_ID` that can be used to override the `GlobalConfiguration().user_id`.

If I understand correctly, you're just trying to override the server identity as reported by the server (`server_info.id`) with the tenant ID when the server is deployed as a cloud tenant:

* use the `ZENML_SERVER_EXTERNAL_SERVER_ID` environment variable (or call `server_config().external_server_id` if set and if running on server), instead of this new `ENV_ZENML_DEPLOYMENT_ID` env var. This takes care of newly deployed tenants.
* for existing tenants, you can either manually update the IDs in the database to retrospectively match their tenant IDs, or you can also patch `SQLZenStore.get_deployment_id()` to use the same override. I prefer the former, because it keeps things nice and consistent and persists all information in the DB. Also easier to apply without upgrades.

Please note that this change is disruptive in many ways, because the deployment ID is used in quite a few places:

* all existing API tokens are invalidated (users will have to re-login, running pipelines will fail unless they use service accounts)
* obviously, it's like a complete change of identity, so new telemetry events will be disconnected from the old ones, unless some other means are used to group them

Reply:
I agree we should probably reuse the `ZENML_SERVER_EXTERNAL_SERVER_ID` env variable. But even if doing that, my question remains the same: Should I just use it once when the DB is first initialized and from that point on use the value stored in the `Identity` table of the database. Or should this env variable be checked anytime the `server_info.id` is checked?